### ISSUE TITLE
Atomic scalar list operations

### ISSUE THREAD

**Issue by pizzafox** [prisma/prisma#6250]
## Problem

> [Scalar lists only support setting a value](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-advanced-types#working-with-scalar-lists--scalar-arrays:~:text=Scalar%20lists%20only%20support%20setting%20a%20value)

This results in a less pleasant developer experience as well as potential data loss from race conditions.

## Suggested solution

Add `push` (add elements) and `pull` (remove elements) operators.
Maybe also a `pop` operator.

Some thought should be put into the syntax used to push 1 element or 1+ elements. [MongoDB uses an object with the `$each` operator](https://docs.mongodb.com/manual/reference/operator/update/push/#append-multiple-values-to-an-array):

```js
db.students.update(
   { name: "joe" },
   { $push: { scores: { $each: [ 90, 92, 85 ] } } }
)
```

## Additional context

Database documentation for array updates:

- [MongoDB](https://docs.mongodb.com/manual/reference/operator/update-array/)
- [PostgreSQL](https://www.postgresql.org/docs/13/arrays.html#ARRAYS-MODIFYING)

**Comment by pizzafox** [comment]
Just realized this is a (partial?) duplicate of #5078 and is partially solved by prisma/prisma-engines#1753.

**Comment by pantharshit00** [comment]
I will consider it as full duplicate of #5078 and it will available in the next release. For deletion, you will need to use set as you can't uniquely identify them and there can be duplicates so it will be application specific logic.
